### PR TITLE
Add C++ Solution + Short Editorial to Time is Mooney (Gold) 

### PR DESCRIPTION
**Pull request by ryanchou-dev**
yay a new pr! 

a lot of sols in intro to dp don't have cpp solutions, so ill be adding some throughout this week :)

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-993.mdx [hunk 1]
```diff
@@ -2,17 +2,78 @@
...
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAX_DAYS = 1000;
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:24]: i see you learned, nice

File: solutions/usaco-993.mdx [hunk 2]
```diff
@@ -2,17 +2,78 @@
...
+	for (int i = 0; i < m; i++) {
+		int u, v;
+		cin >> u >> v;
+		u--;
+		v--;
+
+		adj[u].push_back(v);
+	}
+
+	// dp[i][j] = the maximum money that Bessie can make on day i
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:49]: where does j come into play then?

File: solutions/usaco-993.mdx [hunk 3]
```diff
@@ -2,17 +2,78 @@
...
+
+	vector<vector<int>> adj(n);
+
+	for (int i = 0; i < m; i++) {
+		int u, v;
+		cin >> u >> v;
+		u--;
+		v--;
+
+		adj[u].push_back(v);
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:46]:
```suggestion
		adj[--u].push_back(--v);
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-993.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-993.mdx:49]: j's the city, ill add that in

File: solutions/usaco-993.mdx [hunk 4]
```diff
@@ -43,7 +43,7 @@ int main() {
 		adj[--u].push_back(--v);
 	}
 
-	// dp[i][j] = the maximum money that Bessie can make on day i
+	// dp[i][j] = the maximum money that Bessie can make on day i in city j
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:46]: sorry lol
```suggestion
	// dp[i][j] = the max money that Bessie can make on day i if she ends in city j
```

File: solutions/usaco-993.mdx [hunk 5]
```diff
@@ -2,17 +2,72 @@
...
 source: USACO Gold 2020 January
 title: Time is Mooney
-author: Nathan Gong
+author: Nathan Gong, Ryan Chou
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_time_gold_jan20.html)
 
+
+We only need to keep track of the number of days that Bessie has been traveling and the amount of mooney that she has made. For every day up to $T_{\text{max}}$, we'll try to travel to each city, and update the mooney made if the current path is more optimal.
```
Comment (SansPapyrus683) [inline solutions/usaco-993.mdx:11]: maybe explain how T_max was calculated?

**Review by SansPapyrus683 — APPROVED** [review]
bing chilling

**Review by nwatx — APPROVED** [review]
looks excellent

**Review by TruaShamu — APPROVED** [review]
looks good to me :)

**Comment by SansPapyrus683** [comment]
2 mods & another content author

**Pull request merged**